api/src/main/java/io/opencensus/resource/Resource.java

```diff
@@ -0,0 +1,181 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opencensus.resource;
+
+import com.google.auto.value.AutoValue;
+import io.opencensus.common.ExperimentalApi;
+import io.opencensus.internal.DefaultVisibilityForTesting;
+import io.opencensus.internal.StringUtils;
+import io.opencensus.internal.Utils;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.Immutable;
+
+/**
+ * {@link Resource} represents a resource, which capture identifying information about the entities
+ * for which signals (stats or traces) are reported. It further provides a framework for detection
+ * of resource information from the environment and progressive population as signals propagate from
+ * the core instrumentation library to a backend's exporter.
+ *
+ * @since 0.18
+ */
+@Immutable
+@AutoValue
+@ExperimentalApi
+public abstract class Resource {
+  @DefaultVisibilityForTesting static final int MAX_LENGTH = 255;
+  private static final String OC_RESOURCE_TYPE_ENV = "OC_RESOURCE_TYPE";
+  private static final String OC_RESOURCE_LABELS_ENV = "OC_RESOURCE_LABELS";
+  private static final String LABEL_LIST_SPLITTER = ",";
+  private static final String LABEL_KEY_VALUE_SPLITTER = "=";
+  private static final String ERROR_MESSAGE_INVALID_CHARS =
+      " should be a ASCII string with a length greater than 0 and not exceed "
+          + MAX_LENGTH
+          + " characters.";
+  private static final String ERROR_MESSAGE_INVALID_VALUE =
+      " should be a ASCII string with a length not exceed " + MAX_LENGTH + " characters.";
+
+  @Nullable
+  private static final String ENV_TYPE = parseResourceType(System.getenv(OC_RESOURCE_TYPE_ENV));
+
+  private static final Map<String, String> ENV_LABEL_MAP =
+      parseResourceLabels(System.getenv(OC_RESOURCE_LABELS_ENV));
+
+  Resource() {}
+
+  /**
+   * Returns the type identifier for the resource.
+   *
+   * @return the type identifier for the resource.
+   * @since 0.18
+   */
+  @Nullable
+  public abstract String getType();
+
+  /**
+   * Returns a map of labels that describe the resource.
+   *
+   * @return a map of labels.
+   * @since 0.18
+   */
+  public abstract Map<String, String> getLabels();
+
+  /**
+   * Returns a {@link Resource}. This resource information is loaded from the OC_RESOURCE_TYPE and
+   * OC_RESOURCE_LABELS environment variables.
+   *
+   * @return a {@code Resource}.
+   * @since 0.18
+   */
+  public static Resource create() {
```

Comment:
Probably this should be named as "createFromEnvironmentVariables".

Reply:
Done in https://github.com/census-instrumentation/opencensus-java/pull/1565/commits/1fae2007a75b575920ebc7ff4da90dd627e22059. PTAL